modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java

```diff
@@ -0,0 +1,266 @@
+   * that the correspondent {@link ValuesProvider} will require a connection.
+   *
+   * @param modelPropertyBuilder     Options Resolver Model Property Builder
+   * @param parameterizableComponent component to introspect
+   */
+  private void enrichWithConnection(ValuesProviderFactoryModelPropertyBuilder modelPropertyBuilder,
+                                    ParameterizableTypeWrapper parameterizableComponent) {
+    List<FieldElement> connectionFields = parameterizableComponent.getAnnotatedFields(Connection.class);
+    if (!connectionFields.isEmpty()) {
+      FieldElement fieldElement = connectionFields.get(0);
+      modelPropertyBuilder.withConnection(fieldElement.getField());
+    }
+  }
+
+  /**
+   * Introspects the given {@link ParameterizableTypeWrapper parameterizableComponent} looking if this ones uses a
+   * {@link Config}, if this is true this method will indicate to the {@link ValuesProviderFactoryModelPropertyBuilder}
+   * that the correspondent {@link ValuesProvider} will require a config.
+   *
+   * @param modelPropertyBuilder     Options Resolver Model Property Builder
+   * @param parameterizableComponent component to introspect
+   */
+  private void enrichWithConfiguration(ValuesProviderFactoryModelPropertyBuilder modelPropertyBuilder,
+                                       ParameterizableTypeWrapper parameterizableComponent) {
+    List<FieldElement> configFields = parameterizableComponent.getAnnotatedFields(Config.class);
+    if (!configFields.isEmpty()) {
+      FieldElement fieldElement = configFields.get(0);
+      modelPropertyBuilder.withConfig(fieldElement.getField());
+    }
+  }
+
+  /**
+   * Given a list of {@link ParameterDeclaration}, introspect it and finds all the considered parameters with an associated
+   * {@link ValuesProvider}
+   *
+   * @param parameters parameters to introspect
+   * @return a Map containing all the {@link ParameterDeclaration} with their correspondent {@link ValuesProvider}
+   */
+  private Map<ParameterDeclaration, Class<? extends ValuesProvider>> introspectParameters(List<ParameterDeclaration> parameters) {
+
+    Map<ParameterDeclaration, Class<? extends ValuesProvider>> optionResolverEnabledParameters = new HashMap<>();
+
+    parameters.forEach(param -> getAnnotation(param, OfValues.class)
+        .ifPresent(optionAnnotation -> optionResolverEnabledParameters.put(param, optionAnnotation.value())));
+
+    return optionResolverEnabledParameters;
+  }
+
+  /**
+   * Given a list of {@link ParameterGroupDeclaration}, introspect it and finds all the considered parameters with an associated
+   * {@link ValuesProvider}
+   *
+   * @param parameterGroups parameter groups to introspect
+   * @return a Map containing all the {@link ParameterGroupDeclaration} with their correspondent {@link ValuesProvider}
+   */
+  private Map<ParameterGroupDeclaration, Class<? extends ValuesProvider>> introspectParameterGroups(List<ParameterGroupDeclaration> parameterGroups) {
```

Comment:
is not required to keep this introspection logic, but because the way that multi-level are represented this eases a lot the enricher logic.